pkg/controller/sync/controller.go

```diff
@@ -328,13 +328,9 @@ func (s *FederationSyncController) syncToClusters(fedResource FederatedResource,
 			clusterObj = rawClusterObj.(*unstructured.Unstructured)
 		}
 
-		if clusterObj != nil && fedResource.SkipClusterChange(clusterObj) {
-			continue
-		}
-
 		// Resource should not exist in the named cluster
 		if !selectedClusterNames.Has(clusterName) {
-			if clusterObj != nil {
+			if clusterObj != nil && !fedResource.SkipClusterChange(clusterObj) {
```

Comment:
Github review does not allow to give comment on most of the portion of unchanged code.
The comments in the definition of this function might need a little change now.

```
        // The Namespace type is a special case because it is the only
	// container in the Kubernetes API.  Federation presumes a
	// separation between the template and target resources, but a
	// namespace in the host cluster is necessarily both template and
	// target.
```

Reply:
Good catch.  I've changed the name of this function to `SkipClusterDelete` and updated the documentation accordingly (including the block deleted from `ObjectForCluster`).